src/Subtitles/SubtitleComponent.js

```diff
@@ -0,0 +1,203 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Lightning from '../Lightning'
+
+export default class SubtitleComponent extends Lightning.Component {
+  static _template() {
+    return {
+      visible: false,
+      rect: true,
+      color: 0x90000000,
+      shader: { type: Lightning.shaders.RoundedRectangle, radius: 5 },
+      Text: {
+        y: 5,
+        x: 20,
+        text: {
+          textColor: 0xffffffff,
+          fontSize: 38,
+          lineHeight: 38 * 1.4,
+          textAlign: 'center',
+          wordWrap: true,
+          maxLines: 3,
+          shadow: true,
+          shadowColor: 0xff333333,
+        },
+      },
+    }
+  }
+
+  _init() {
+    this._textTextureDefaults = new Lightning.textures.TextTexture(this.stage).cloneArgs()
+
+    this.tag('Text').on('txLoaded', ({ _source }) => {
+      this.w = _source.w + this.tag('Text').x * 2
+      this.h = _source.h
+      this.position()
+    })
+  }
+
+  get textFormat() {
+    const textTag = this.tag('Text').text
+    return {
+      fontFace: textTag.fontFace || 'sans-serif',
+      fontSize: textTag.fontSize,
+      lineHeight: textTag.lineHeight,
+      textAlign: textTag.textAlign,
+      wordWrap: true,
+      maxLines: textTag.maxLines,
+    }
+  }
+
+  show() {
+    this.visible = true
+  }
+
+  hide() {
+    this.visible = false
+  }
+
+  position() {
+    this.x = this._calculateX(this.xPos)
+    this.y = this._calculateY(this.yPos)
+  }
+
+  set viewportW(v) {
+    this._viewportW = v
+    this.x = this._calculateX(this.xPos)
+  }
+
+  get viewportW() {
+    return this._viewportW || this.application.finalW
+  }
+
+  set viewportH(v) {
+    this._viewportH = v
+    this.y = this._calculateY(this.yPos)
+  }
+
+  get viewportH() {
+    return this._viewportH || this.application.finalH
+  }
+
+  _calculateX(x) {
+    if (x === 'center') {
+      x = (this.viewportW - this.finalW) / 2
+    } else if (x === 'left') {
+      x = 60
+    } else if (x === 'right') {
+      x = this.viewportW - this.finalW - 60
+    }
+    return x
+  }
+
+  set xPos(v) {
+    this._x = v
+    this.x = this._calculateX(v)
+  }
+
+  get xPos() {
+    return this._x || 'center'
+  }
+
+  _calculateY(y) {
+    if (y === 'center') {
+      return (this.viewportH - this.finalH) / 2
+    } else if (y === 'top') {
+      return 60
+    } else if (y === 'bottom') {
+      return this.viewportH - this.finalH - 60
+    }
+    return y
+  }
+
+  set yPos(v) {
+    this._y = v
+    this.y = this._calculateY(v)
+  }
+
+  get yPos() {
+    return this._y || 'bottom'
+  }
+
+  set fontFamily(v) {
+    this.tag('Text').text.fontFace = v
+  }
+
+  set fontSize(v) {
+    this.tag('Text').text.fontSize = v
+    this.tag('Text').text.lineHeight = v * 1.3
+  }
+
+  set fontColor(v) {
+    this.tag('Text').color = v
+  }
```

Comment:
Should this also be renamed to textColor? And it should also patch the textColor, not the color property of the Texture.

Reply:
It's actually less CPU intensive to change the color property of the texture versus setting the textColor of the text itself. I don't think there is a real need to change the name of the setter either.

Reply:
That's actually a great tip that never crossed my mind! 👍

Reply:
However, with this in mind, should the initial state also use the color and not the textColor property of the texture?

https://github.com/rdkcentral/Lightning-SDK/blob/246e700e4c19c7f0ca1f22fb255df2223fb0d721/src/Subtitles/SubtitleComponent.js#L29-L42

Reply:
We could technically omit the initial textColor, since it default to white anyway, but I don't mind having it there for clarity reasons. The technique used (setting the color instead of the textColor) means that we are tinting the text Texture with the color. And since the text itself is white it will result in the exact color that we pass it (tinting a gray text texture would have a different result for example)